### PR TITLE
DEVOPS-725/incident severity definitions

### PR DESCRIPTION
**Pull request by kschaer**
# Summary
Defines severity levels for incidents at Fabric.

This PR seeks to establish:
- what are severity levels
- why are severity levels useful? how do they help us respond to incidents more effectively?
- High level definitions for SEV-1 through SEV-5 severity incidents
- More specific definitions and examples, which are more closely tied to Fabric's business and products
- At a high level, what type of response/mobilization is required at each severity level? 

## How to successfully review this PR
If you have not already done so, please take some time to gain some basic familiarity with this repository. It was forked directly from PagerDuty's repo (which is also [published to their site here](https://response.pagerduty.com/), which may provide a more pleasant reading experience). Familiarity with that content will be a big help in reviewing the work as we adopt this content for Fabric.

I am intentionally keeping these PR's fairly small in scope. The file(s) touched in this PR may link out to other files within the repository that have not yet been adapted for Fabric's use.

Any and all feedback is welcome, including direct commits to this branch if you would like. If there are other members of the Fabric team you feel would provide important feedback here, please feel free to pull them into the convo.

### CODE REVIEW

File: docs/before/severity_levels.md [hunk 1]
```diff
@@ -1,15 +1,36 @@
...
 
 !!! tip "Always Assume The Worst"
      If you are unsure which level an incident is (e.g. not sure if SEV-2 or SEV-1), **treat it as the higher one**. During an incident is not the time to discuss or litigate severities, just assume the highest and review during a post-mortem.
 
 !!! question "Can a SEV-3 be a major incident?"
      All SEV-2's are major incidents, but not all major incidents need to be SEV-2's. If you require co-ordinated response, even for lower severity issues, then trigger our incident response process. The IC can make a determination on whether full incident response is necessary.
 
+## Severity Definitions
+
 <table class="custom-table">
```
Comment (kschaer) [inline docs/before/severity_levels.md:34]: I'm not thinking too hard about the structure of this doc; the table format pagerduty gave us here feels fine for now.

File: docs/before/severity_levels.md [hunk 2]
```diff
@@ -24,18 +45,21 @@ The first step in any incident response process is to determine what actually [c
       <td>
         <p class="intent">Critical issue that warrants public notification and liaison with executive teams.</p>
         <ul>
-          <li>The system is in a critical state and is actively impacting a large number of customers.</li>
-          <li>Functionality has been severely impaired for a long time, breaking SLA.</li>
-          <li>Customer-data-exposing security vulnerability has come to our attention.</li>
+          <li>Fabric's site or mobile app is in a critical state and is actively impacting a large number of customers.</li>
```
Comment (tamwahba) [inline docs/before/severity_levels.md:48]: do we define large in a certain way here?

File: docs/before/severity_levels.md [hunk 3]
```diff
@@ -44,17 +68,19 @@ The first step in any incident response process is to determine what actually [c
...
         <ul>
-          <li>Notification pipeline is severely impaired.</li>
-          <li>Incident response functionality (ack, resolve, etc) is severely impaired.</li>
-          <li>Web app is unavailable or experiencing severe performance degradation for most/all users.</li>
-          <li>Monitoring of PagerDuty systems for major incident conditions is impaired.</li>
-          <li>Any other event to which a PagerDuty employee deems necessary of incident response.</li>
+          <li>The critical path to purchase an insurance product is severely impaired.</li>
+          <li>The critical path to create a will or obtain another financial product is severely impaired.</li>
+          <li>Underwriters cannot access or perform necessary underwriting tasks on many applications in Loom.</li>
+          <li>Web app or mobile app is unavailable or experiencing severe performance degradation for most/all users.</li>
```
Comment (tamwahba) [inline docs/before/severity_levels.md:74]: so compared to SEV-1 above
> Fabric's site or mobile app is in a critical state and is actively impacting a large number of customers.
performance degradation here means partial access? as opposed to critical state which is no access at all?

File: docs/before/severity_levels.md [hunk 4]
```diff
@@ -113,12 +141,16 @@ The first step in any incident response process is to determine what actually [c
...
+          <li>Create a JIRA ticket and assign to owner of the affected service.</li>
+          <li>Post in the team slack channel relevant to the affected service, informing the team of the issue and linking them to the ticket.</li>
         </ul>
       </td>
     </tr>
   </tbody>
 </table>
 
-!!! note "Be Specific"
-    These severity descriptions have been changed from the PagerDuty internal definitions to be more generic. For your own documentation, you are encouraged to make your definitions very specific, usually referring to a % of users/accounts affected. You will usually want your severity definitions to be metric driven.
```
Comment (tamwahba) [inline docs/before/severity_levels.md:124]: ok so it seems like this is following a template from pagerduty? I feel like the part about more specific definitions around what's large/critical/degraded is useful to apply here

**Review by tamwahba — COMMENTED** [review]
A few questions

File: docs/before/severity_levels.md (hunk 2 above)
Comment (kschaer) [reply docs/before/severity_levels.md]: I think at this iteration, the distinction I'm trying to make is generally "widespread" vs "edge case". I'll try and make this more explicit.

File: docs/before/severity_levels.md [hunk 5]
```diff
@@ -1,15 +1,43 @@
...
+- Is this a security incident?
+- What level of risk-taking is appropriate for remediation of this issue?
+- Do we need to inform the public of a problem with Fabric, or contact specific customers?
+- Do I need to involve internal stakeholders, such as Marketing or Compliance?
+
+Agreeing upon the severity level of an incident helps all incident responders, and the Fabric organization, get on the same page about the incident response process. Remember: none of this happens in a vacuum. If you are unsure how to answer any of these core questions, don't hesitate to involve the incident commander.
+
+### Don't get hung up on severity levels during an active incident
+
+While our severity definitions can provide a helpful starting place for mobilizing the right level of incident response, it's important not to get too hung up on exactly what the severity level is. When in doubt, it's worth rounding up to the higher severity level. It's always better to mobilize a larger response at first and then scale down as necessary, versus having an inadequate response (and potentially leading the issue to worsen over time).
```
Comment (liamfd) [inline docs/before/severity_levels.md:22]: Nit: this seems somewhat duplicative of the "always assume the worst" tip, maybe we could fold the "scaling up/down" part into the tip?

File: docs/before/severity_levels.md (hunk 3 above)
Comment (kschaer) [reply docs/before/severity_levels.md]: Good call out, these were definitely too similar. I agree that "critical state" is not well defined, too. I'm not attached to that language however. I guess what I'm trying to distinguish here is the difference between a full outage, or loss of functionality comparable to an outage, VS poor performance and experience but the site is still generally available and core functionality is not busted (other lines describe situations where core functionality is indeed busted). 

I've tried to make these levels more distinct, with the relevant SEV-1 definition being:
```
Fabric's site or mobile app is unavailable or has widespread loss of functionality, and is actively impacting a large number of customers. This is often described as an outage.
```

and SEV-2 being:
```
Fabric's site or mobile app is experiencing performance degradation for most/all users, but overall functionality is still available.
```

**Review by liamfd — APPROVED** [review]
One nit, otherwise LGTM!

File: docs/before/severity_levels.md (hunk 4 above)
Comment (kschaer) [reply docs/before/severity_levels.md:124]: I agree, but I don't think I have what we need to add metrics to these definitions quite yet. I've added some notes and TODO's above the table of severity levels to this end. 

On the other hand however, I think that many metrics for incidents / severity levels will be best owned/defined by individual Fabric services. I've tried to keep things pretty "global" here - tying the definitions to Fabric's core functionality, VS the individual services that support that functionality. I'd love to provide a template/scaffold for defining severity levels in individual services eventually, and have added a TODO for this work.

File: docs/before/severity_levels.md (hunk 5 above)
Comment (kschaer) [reply docs/before/severity_levels.md:22]: I suppose this is just a point I wanted to emphasize - I agree that there's some overlap.

**Pull request closed without merging**